deploy/kubedirector/kubedirector.hpe.com_kubedirectorapps_crd.yaml

```diff
@@ -4,198 +4,386 @@ metadata:
   name: kubedirectorapps.kubedirector.hpe.com
 spec:
   group: kubedirector.hpe.com
-  version: v1beta1
-  names:
-    kind: KubeDirectorApp
-    listKind: KubeDirectorAppList
-    plural: kubedirectorapps
-    singular: kubedirectorapp
-    shortNames:
-      - kdapp
-  scope: Namespaced
-  validation:
-    openAPIV3Schema:
-      type: object
-      required: [apiVersion, kind, metadata, spec]
-      properties:
-        apiVersion:
-          type: string
-        kind:
-          type: string
-        metadata:
-          type: object
-        spec:
-          type: object
-          required: [label, distroID, version, roles, config, configSchemaVersion]
-          properties:
-            label:
-              type: object
-              required: [name]
-              properties:
-                name:
-                  type: string
-                  minLength: 1
-                description:
-                  type: string
-            distroID:
-              type: string
-              minLength: 1
-            version:
-              type: string
-              minLength: 1
-            configSchemaVersion:
-              type: integer
-              minimum: 7
-            defaultImageRepoTag:
-              type: string
-              minLength: 1
-            defaultConfigPackage:
-              type: object
-              nullable: true
-              required: [packageURL]
-              properties:
-                packageURL:
-                  type: string
-                  pattern: '^(file|https?)://.+\.tgz$'
-            services:
-              type: array
-              items:
+  preserveUnknownFields: false
+  versions:
+  - name: v1beta1
+    served: true
+    storage: true
```

Comment:
Shouldn't beta2 be the storage version?

Reply:
Hmm maybe not. However we would have to be careful not to lose the new v1beta2 fields when the object is stored. I'm reading over https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api_changes.md at the moment and we may want to discuss some aspects of that later.